### PR TITLE
[FSDE-30490] Add support for selecting all items on data table

### PR DESCRIPTION
**Pull request by josecarlosgonzalezv**
https://ctl.atlassian.net/browse/FSDE-30490

### CODE REVIEW

**Review by SergioQCL — APPROVED** [review]

File: src/chi-vue/src/components/data-table/DataTable.tsx [hunk 1]
```diff
@@ -631,6 +637,37 @@ export default class DataTable extends Vue {
     return this.selectedRows.includes(row.rowId);
   }
 
+  _selectAllDropdown() {
+    const pageSelectionMessage = 'Select all items, this page';
+    const allPagesSelectionMessage = 'Select all items, all pages';
+    const deselectAllSelectionMessage = 'Deselect none';
```
Comment (mattnickles) [inline src/chi-vue/src/components/data-table/DataTable.tsx:643]: Sorry there was a typo in the design prototype. This should be 'Deselect all'
```suggestion
    const deselectAllSelectionMessage = 'Deselect all';
```

File: src/chi-vue/src/components/data-table/data-table.scss [hunk 2]
```diff
@@ -110,6 +101,14 @@ $font-weight-semi-bold: 600;
         position: relative;
         text-align: start;
         white-space: normal;
+        
+        &.-selectable {
+          .chi-dropdown {
+            position: absolute;
+            left: 40px;
```
Comment (jllr) [inline src/chi-vue/src/components/data-table/data-table.scss:108]:
```suggestion
            left: 2.5rem;
            position: absolute;
```

File: src/chi/components/data-table/data-table.scss [hunk 3]
```diff
@@ -528,6 +528,14 @@ $bg-url: $bg-url + ' fill="%230262b9" fill-rule="evenodd"/%3E%3C/svg%3E';
 
           .chi-data-table__cell {
             border-bottom: 0.0625rem solid $color-border-base-dark;
+
+            &.-selectable {
+              .chi-dropdown {
+                position: absolute;
+                left: 40px;
```
Comment (jllr) [inline src/chi/components/data-table/data-table.scss:535]:
```suggestion
                left: 2.5rem;
                position: absolute;
```

**Review by jllr — CHANGES_REQUESTED** [review]

**Review by mattnickles — COMMENTED** [review]
@josecarlosgonzalezv @jllr Let's change "Items selected" to "Items Selected"  so the casing is consistent with the title casing used in the "Show Selected Only" checkbox, table headers, etc.

**Review by mattnickles — COMMENTED** [review]
@josecarlosgonzalezv @jllr Do you want to fix the vertical alignment of the items selected label in this PR or separate? We just need to add `align-items: center;` to line 69 of bulk-actions.scss

**Review by mattnickles — APPROVED** [review]

**Pull request merged**